src/code/circuit_python/setup.md

```diff
@@ -0,0 +1,86 @@
+## Installing the Circuit Python Runtime
+
+In order for your microcontroller to be able to use Circuit Python, you
+have to install board-specific firmware.
+
+1. Download the Circuit Python runtime (.uf2 file) for the Raspberry Pi
+PICO from
+[https://circuitpython.org/board/raspberry_pi_pico/](https://circuitpython.org/board/raspberry_pi_pico/).
+
+1. Hold your Gizmo's student processor BOOTSEL ("Boot Select") button down
+while connecting the USB programming cable to your computer. Once you've
+connected the cable, release the BOOTSEL button.
+
+   You should now see an external drive on your computer called "RPI-RP2".
+
+1. Copy the .uf2 file you downloaded to the "RPI-RP2" drive.
+
+   Your device will automatically reboot and reconnect as a new drive
+   called "CIRCUITPY".
+
+If your Gizmo ever gets stuck in a way that prevents you from saving new
+Circuit Python code to it or if it stops showing up as the CIRCUITPY drive
+when you connect it, you may need to re-install the runtime by following
+the above steps again.
+
+## Installing the Mu Editor
+
+[Mu](https://codewith.mu/) is a simple Python editor that is the
+recommended editor for Circuit Python code.
+
+If you would like more information about using the Mu editor, including an
+introduction to the user interface, checkout the
+[Mu tutorial articles](https://codewith.mu/en/tutorials).
+
+1. Download and run the installer for your operating system from
+   [the Mu download page](https://codewith.mu/en/download).
+
+1. When prompted for what mode to run Mu in, make sure to select Circuit
+   Python.
+
+   ![Mu mode selection screenshot](https://cdn-learn.adafruit.com/assets/assets/000/105/681/medium640/circuitpython_WtCP_Mu_mode_dialogue.png?1634750676)
+
+1. Mu tries to automatically connect to Circuit Python devices attached to
+   your computer. By default, when you save the code you're writing in Mu,
+   it will save directly to your Circuit Python device. If it can't find
+   one, it will show you this warning explaining where the code you write
+   will be saved on your computer.
+
+   ![Mu device not found screenshot](https://cdn-learn.adafruit.com/assets/assets/000/105/679/medium640/circuitpython_WtCP_Mu_device_not_found.png?1634749722)
+
+## Installing the CircUp Library Manager
+
+Like your code, libraries for Circuit Python devices are managed by
+copying files onto the CIRCUITPY drive. While this can be done manually,
+it involves quite a bit of downloading files from different websites and
+extracting zipped folders. The CircUp tool manages these processes for you
+to keep your installed libraries up to date.
+
+Adafruit provides detailed installation instructions for Windows, Mac, and
+Linux. You can find these instructions
+[here](https://learn.adafruit.com/keep-your-circuitpython-libraries-on-devices-up-to-date-with-circup).
+Specifically, follow the steps in the "Prepare for Install" and "Install
+CircUp" sections.
+
+## Installing the Gizmo Circuit Python Library
```

Comment:
This section doesn't appear to talk about versions and should.  In general installing an un-pinned version is not something we want to encourage people to do.

Reply:
I can specify the CircuitPython version to be 9.X to make sure they have a compatible runtime in case a new major release comes out.
For the Gizmo library, is there any reason we wouldn't want them to use the latest release?

Reply:
In general my intent is to qualify a specific major release at the start of a competition season and then pin that.  Patch releases on top are fine as long as they don't break compat, but there should probably be a notice either here or in the repo about updates and the safety thereof.

Reply:
I've added a callout to make sure they're using the correct major+minor version of the Circuit Python runtime firmware.
Unfortunately, it doesn't look like circup has support for pinning module versions, so it always grabs the latest version. I suspect this won't be a real issue for us this year. There's perhaps an interesting conversation to be had about whether we call out specific versions in documentation like this or just freeze major releases during competition seasons such that grabbing latest always gives the official version for that season.

Reply:
What an odd limitation.  If they don't use circup it would use the specific bundle we provide right?

Reply:
Yes, when manually managing libraries / bundles we can specify whatever version we want.
As mentioned at dinner tonight, I can add a section to these docs later that spells out more of the details of manual library management.